application/common/models/User.php

```diff
@@ -149,6 +155,26 @@ public function beforeDelete()
         return true;
     }
 
+    // The MfaBackend code for deleting a webauthn type MFA
+    // isn't designed to do this so easily.
+    private static function deleteWebauthns(MFA $mfa)
+    {
+        $children = $mfa->mfaWebauthns;
+        foreach ($children as $child) {
+            if (!$child->delete()) {
+                \Yii::error([
+                    'action' => 'delete mfa webauthn child record before deleting mfa',
+                    'status' => 'error',
+                    'error' => $child->getFirstErrors(),
+                    'mfa_id' => $mfa->id,
+                    'child_id' => $child->id,
+                ]);
+                return false;
+            }
+        }
```

Comment:
This is essentially the code that would be moved to the MfaWebauthn->beforeDelete() method, per my other comment.